lib/index.js

```diff
@@ -36,37 +36,38 @@ export default function runContentfulExport (usageParams) {
 
         log.info('Downloading ' + response.assets.length + ' assets')
 
-        var arrayOfDownloadPromises = []
-        response.assets.forEach((asset) => {
+        Promise.map(response.assets, (asset) => {
           if (asset.fields.file) {
-            Object.keys(asset.fields.file).forEach((key) => {
-              arrayOfDownloadPromises.push(
-                downloadAsset(asset.fields.file[key].url, opts.exportDir).then(
-                  (downLoadedFile) => {
+            return Object.keys(asset.fields.file).reduce((chain, key) => {
```

Comment:
Why not something like:

```js
const keys = Object.keys(...);

Promise.map(keys, () => {
   fetchTheAsset
   .then(foo)
   .catch(bar)
})
```

I think it might be easier to readh